dns_reverse_proxy.go

```diff
@@ -21,46 +22,68 @@ package main
 
 import (
 	"flag"
+	"fmt"
 	"log"
+	"math/rand"
 	"net"
 	"os"
 	"os/signal"
 	"strings"
 	"syscall"
+	"time"
 
 	"github.com/miekg/dns"
 )
 
+type stringArrayFlags []string
+
+func (i *stringArrayFlags) String() string {
+	return fmt.Sprint(*i)
+}
+
+func (i *stringArrayFlags) Set(value string) error {
+	*i = append(*i, value)
+	return nil
+}
+
 var (
 	address = flag.String("address", ":53", "Address to listen to (TCP and UDP)")
 
 	defaultServer = flag.String("default", "",
 		"Default DNS server where to send queries if no route matched (host:port)")
 
-	routeList = flag.String("route", "",
-		"List of routes where to send queries (domain=host:port)")
-	routes map[string]string
+	routeLists stringArrayFlags
+	routes     map[string][]string
 
 	allowTransfer = flag.String("allow-transfer", "",
 		"List of IPs allowed to transfer (AXFR/IXFR)")
 	transferIPs []string
 )
 
 func main() {
+	rand.Seed(time.Now().Unix()) // initialize global pseudo random generator
```

Comment:
Can you move these two lines into a separate `init()`?
And drop the comment on `rand.Seed`, it's clear what it does (to me at least).

Reply:
ok